src/expression.c

```diff
@@ -118,7 +118,12 @@ Expression *getRightThis(Loc loc, Scope *sc, AggregateDeclaration *ad,
                     {
                         //printf("rewriting e1 to %s's this\n", f->toChars());
                         n++;
+#ifdef IN_GCC
+                        if (n > 1)
+                            e1 = new VarExp(loc, f->vthis);
+#else
                         e1 = new VarExp(loc, f->vthis);
+#endif
```

Comment:
Here's the GDC-glue fix: https://github.com/D-Programming-GDC/GDC/commit/84e77850a4e9184526d9b571fb8bdfda4dc103a1